src/ray/raylet/scheduling/fair_scheduling_queue.h

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+/* #include <tuple> */
+#include "ray/common/task/task.h"
+#include "ray/rpc/node_manager/node_manager_server.h"
+
+namespace ray {
+namespace raylet {
+
+/// Work represents all the information needed to make a scheduling decision.
+/// This includes the task, the information we need to communicate to
+/// dispatch/spillback and the callback to trigger it.
+using Work = std::tuple<Task, rpc::RequestWorkerLeaseReply *, std::function<void(void)>>;
+
+namespace internal {
+
+struct SchedulingPriorityComparator : public std::less<SchedulingClass> {
+ public:
+  explicit SchedulingPriorityComparator(
+      std::unordered_map<SchedulingClass, uint64_t> &priorities);
+
+  bool operator()(const SchedulingClass &lhs, const SchedulingClass &rhs) const;
+
+ private:
+  /// A reference to FairSchedulingQueue::active_works_
+  std::unordered_map<SchedulingClass, uint64_t> &priorities_;
```

Comment:
Why don't we just call it `active_works_`?  I think it is okay to expose impl details in this case.